### PR TITLE
Reports support force height

### PR DESCRIPTION
**Pull request by yanyo87**
related <https://github.com/demisto/etc/issues/33270#issuecomment-817542168>
Related <https://github.com/demisto/etc/issues/35454>

1. Add border to widgets 
image

+ tables 
image

2. fix padding legend column chart 
image

3. Add support for force height - column chart & line chart 

4. add support to disable force height for tables 

Force full table - enabled 
image

image

Force full table - disabled 
image

Vertical bars  stacked & unstacked - 

report builder - 
image

after fix - 
image
[localhost.pdf](https://github.com/demisto/sane-reports/files/6323077/localhost.pdf)

before fix -
image

Horizontal bar - 
report builder - 
image

after fix - 

image

image
image

before fix -  
image

_____________

Pie Chart vertical & horizontal 

report builder - vertical 
image

report builder horizontal - 
image

after fix - 
image
image

[localhost - pie.pdf](https://github.com/demisto/sane-reports/files/6323438/localhost.-.pie.pdf)

before fix - 
image
___________
 
report builder Line chart -
image

after fix - 
image
[localhost - line .pdf](https://github.com/demisto/sane-reports/files/6323476/localhost.-.line.pdf)
 
Before  - Line Charts 
image

_____________

Report Builder table 
image

After fix - 
image

image

### CODE REVIEW

File: src/components/Sections/SectionChart/SectionBarChart.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const createXAxisProps = (data, dataKey, width) => {
 };
 
 const SectionBarChart = ({ data, style, dimensions, legend, chartProperties = {},
-  legendStyle = null, sortBy, stacked, referenceLineY }) => {
+  legendStyle = null, sortBy, stacked, referenceLineY, showOverflow }) => {
```
Comment (yanyo87) [inline src/components/Sections/SectionChart/SectionBarChart.js:49]: lets call it 
showFullChart

File: src/components/Sections/SectionChart/SectionBarChart.js [hunk 2]
```diff
@@ -159,18 +159,24 @@ const SectionBarChart = ({ data, style, dimensions, legend, chartProperties = {}
       return item;
     });
   }
+  let isFull = false;
+  if (!showOverflow && (preparedData.length * CHART_LEGEND_ITEM_HEIGHT > dimensions.height || stacked)) {
+    isFull = true;
+    dimensions.height = (preparedData.length * CHART_LEGEND_ITEM_HEIGHT) + 300;
```
Comment (yanyo87) [inline src/components/Sections/SectionChart/SectionBarChart.js:165]: the 300 should be const

File: src/components/Sections/SectionChart/SectionChart.js [hunk 3]
```diff
@@ -45,6 +45,7 @@ const SectionChart = ({ type, data: rawData, style, dimensions, legend, chartPro
                       sortBy={sortBy}
                       stacked={stacked}
                       referenceLineY={referenceLineY}
+                      showOverflow={showOverflow}
```
Comment (yanyo87) [inline src/components/Sections/SectionChart/SectionChart.js:48]: lets call it 
showFullChart or showFullChartLegend

File: src/components/Sections/SectionChart/SectionLineChart.js [hunk 4]
```diff
@@ -151,6 +152,10 @@ const SectionLineChart = ({ data, style, dimensions, legend, chartProperties = {
     });
   }
 
+  if (!showOverflow && preparedLegend.length * CHART_LEGEND_ITEM_HEIGHT > dimensions.height) {
+    dimensions.height = (preparedLegend.length * CHART_LEGEND_ITEM_HEIGHT) + 200;
```
Comment (yanyo87) [inline src/components/Sections/SectionChart/SectionLineChart.js:156]: lets move the 200 to const

File: src/components/Sections/SectionChart/SectionLineChart.less [hunk 5]
```diff
@@ -8,6 +8,10 @@
   }
 
   .recharts-legend-wrapper {
+
+    &:not(.section-show-overflow) {
+      max-height: 100%;
+    }
```
Comment (yanyo87) [inline src/components/Sections/SectionChart/SectionLineChart.less:14]: let move it to -> 
ReportLayout.less

File: src/components/Sections/SectionChart/SectionPieChart.js [hunk 6]
```diff
@@ -86,6 +91,14 @@ const SectionPieChart = ({ data, style, dimensions, legend, chartProperties = {}
     legendHeight = dimensions.height / 2;
   }
 
+  if (!showOverflow && preparedData.length * CHART_LEGEND_ITEM_HEIGHT > dimensions.height) {
+    legendHeight = (preparedData.length * CHART_LEGEND_ITEM_HEIGHT);
+    if (chartProperties.layout === CHART_LAYOUT_TYPE.vertical) {
+      legendHeight += 100;
```
Comment (yanyo87) [inline src/components/Sections/SectionChart/SectionPieChart.js:97]: const

File: src/components/Sections/SectionChart/SectionBarChart.less [hunk 7]
```diff
@@ -11,8 +11,9 @@
 }
 .section-column-chart {
   height: 100%;
+  padding-top: 5px;
```
Comment (Bars92) [inline src/components/Sections/SectionChart/SectionBarChart.less:14]: why would you add padding and negative top?

File: src/components/Sections/SectionChart/SectionBarChart.js (hunk 2 above)
Comment (Bars92) [reply src/components/Sections/SectionChart/SectionBarChart.js:165]: whats the 300 for anyway?

File: src/components/Sections/SectionChart/SectionChart.js [hunk 8]
```diff
@@ -21,7 +21,7 @@ const filterResults = (chartProperties, rawResults) => {
 };
 
 const SectionChart = ({ type, data: rawData, style, dimensions, legend, chartProperties = {}, legendStyle = {}, sortBy,
-  referenceLineX, referenceLineY, title, stacked, fromDate, toDate, titleStyle }) => {
+  referenceLineX, referenceLineY, title, stacked, fromDate, toDate, titleStyle, showOverflow }) => {
```
Comment (Bars92) [inline src/components/Sections/SectionChart/SectionChart.js:24]: who is passing the showOverflow prop?

**Review by Bars92 — COMMENTED** [review]
Please also share more updated screenshots and make sure the table is backward compatible

File: src/components/Sections/SectionChart/SectionChart.js (hunk 8 above)
Comment (ikidon) [reply src/components/Sections/SectionChart/SectionChart.js:24]: reportslayouts to layouts

Comment (ikidon) [reply src/components/Sections/SectionChart/SectionChart.js:24]: regarding your q of "whats the 300 for anyway?" (cant answer directly there), when we want to see the full chart and full legend, we stretch the height to to the full height of the legend and then add extra fixed height for the chart, as agreed by the UX team

File: src/components/Sections/SectionChart/SectionBarChart.js (hunk 2 above)
Comment (ikidon) [reply src/components/Sections/SectionChart/SectionBarChart.js]: extra height of the chart after we calculate the room for the legend

File: src/components/Sections/SectionChart/SectionBarChart.less (hunk 7 above)
Comment (yanyo87) [reply src/components/Sections/SectionChart/SectionBarChart.less:14]: cause we have a white background , 
i need to move up the legend so it won't overlap the top axis value.

File: src/components/Layouts/ReportLayout.less [hunk 9]
```diff
@@ -104,6 +116,25 @@ h1 {
         padding: 5px 10px;
       }
 
+      &.section-show-overflow {
```
Comment (Bars92) [inline src/components/Layouts/ReportLayout.less:119]: please use media screen option to avoid showing scroll in print css

**Review by Bars92 — APPROVED** [review]
Great work @yanyo87 ! everything looks so fresh 📊 📈 💯

**Pull request merged**